check_logging.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+DIFF=$(git diff origin/main...HEAD || true)
+
+if [ -z "$DIFF" ]; then
+  echo "No diff found to analyze."
+  exit 0
+fi
+
+PATTERNS=(
+  ".info("
+  ".infof("
+  ".warn("
+  ".warnf("
+  ".error("
+  ".errorf("
+  ".fatal("
+  ".fatalf("
+  "System.out.print"
+  "System.err.print"
+  ".printStackTrace"
+)
+
+ERRORS=""
+CURRENT_FILE=""
+
+while IFS= read -r line; do
+  # Update the current file if a new diff starts
+  if [[ "$line" =~ ^diff\ --git\ a\/.*\ b\/.* ]]; then
+    CURRENT_FILE=$(echo "$line" | awk '{print $3}' | sed 's/^a\///')
+  fi
+
+  if [[ "$line" =~ ^\+[^+] ]]; then
+    # Ignore lines in the test directories
+    if [[ "$CURRENT_FILE" != *"src/test/"* && "$CURRENT_FILE" != *"testsuite/"* ]]; then
```

Comment:
This also needs to exclude this file itself, otherwise it will always fail :) 

[.github/workflows/check_logging.yml](https://github.com/wildfly/wildfly/pull/18116/files#diff-f65ba3d4fa0016b455b4f0a026ba7cab93315c7db73aa6565a677052292e9b37)

Reply:
Yeah, I haven't added it on purpose so that we can check it's working. I'm excluding this file now.

Reply:
> Yeah, I haven't added it on purpose so that we can check it's working. I'm excluding this file now.

If I were to test this, I would test this in a private fork rather than upstream and would test in actual code rather than this file, but that's just a preference 👍🏼